src/java/org/apache/cassandra/config/DatabaseDescriptor.java

```diff
@@ -906,6 +898,36 @@ else if (conf.max_value_size.toMebibytes() >= 2048)
         logInitializationOutcome(logger);
     }
 
+    @VisibleForTesting
+    static void validateUpperBoundStreamingConfig() throws ConfigurationException
+    {
+        // below 2 checks are needed in order to match the pre-CASSANDRA-15234 upper bound for those parameters which were still in megabits per second
+        if (conf.stream_throughput_outbound.toMegabitsPerSecond() >= Integer.MAX_VALUE)
+        {
+            throw new ConfigurationException("Invalid value of stream_throughput_outbound: " + conf.stream_throughput_outbound.toString(), false);
```

Comment:
nit: It might be helpful to mention why something is invalid (i.e. the value overflows int bounds when converted to megabits) in the error message, but if this is only used for testing, I guess it's not a huge concern.